src/zkevm_specs/state.py

```diff
@@ -400,6 +469,12 @@ def get_keys_compressed_in_order(row: Row) -> List[FQ]:
     if row.is_write == 0 and all_keys_eq(row, row_prev):
         assert row.value == row_prev.value
 
+    # 7. Increment mpt_counter
+    #
+    # When previous row is Storage or Account, increment the mpt_counter by one
+    if row_prev.tag() == Tag.Storage or row_prev.tag() == Tag.Account:
+        assert row.mpt_counter == row_prev.mpt_counter + 1
+
```

Comment:
I've taken the approach to make `mpt_counter` persistent because unfortunately `Tag.Storage` and `Tag.Account` are not contiguous (there's `Tag.CallContext` in the middle).  So now `mpt_counter` is forced to start at 0, and is forced to increase by 1 on Tag.Storage or Tag.Account, and be the same on other Tags (except for Start).